include/qdmi/client.h

```diff
@@ -981,6 +1055,329 @@ void QDMI_job_free(QDMI_Job job);
 
 /** @} */ // end of client_job_interface
 
+/** @defgroup client_telemetrysensor_query_interface \
+ *  QDMI Client Telemetry Sensor Query Interface
+ *  @brief Provides functions to query telemetry sensors.
+ *  @details An telemetry sensor query is a task submitted by a client to a
+ * device for querying telemetry sensors, i.e. temperature or power.
+ *
+ *  The typical workflow for a client telemetry sensor query is as follows:
+ *  - Create an telemetry sensor query with @ref
+ * QDMI_device_create_telemetrysensor_query.
+ *  - Set parameters for the telemetry sensor query with @ref
+ * QDMI_telemetrysensor_query_set_parameter
+ *  - Submit the telemetry query to the device with @ref
+ * QDMI_telemetrysensor_query_submit.
+ *  - Check the status of the telemetry sensor query with @ref
+ * QDMI_telemetrysensor_query_check_status.
+ *  - Wait for the telemetry sensor query to finish with @ref
+ * QDMI_telemetrysensor_query_wait.
+ *  - Retrieve the results of the telemetry sensor query with @ref
+ * QDMI_telemetrysensor_query_get_results.
+ *  - Free the telemetry sensor query with @ref
+ * QDMI_telemetrysensor_query_free when it is no longer used.
+ *
+ *  @{
+ */
+
+/**
+ * @brief A handle for a client-side telemetry sensor query.
+ * @details An opaque pointer to a type defined by the driver that encapsulates
+ * all information about an telemetry sensor query submitted to a device by a
+ * client.
+ * @remark Implementations of the underlying type will want to store the device
+ * handle used to create the telemetry sensor query in the handle to be able
+ * to access the device when needed.
+ * @see QDMI_Device_TelemetrySensor_Query for the device-side the telemetry
+ * sensor query handle.
+ */
+typedef struct QDMI_TelemetrySensor_Query_impl_d *QDMI_TelemetrySensor_Query;
+
+/**
+ * @brief Create an telemetry sensor query.
+ * @details This is the main entry point for a client to submit an telemetry
+ * sensor query to a device. The returned handle can be used throughout the
+ * @ref client_telemetrysensor_query_interface
+ * "client telemetry sensor query interface" to refer to the telemetry
+ * sensor query.
+ * @param[in] device The device to create the telemetry sensor query on. Must
+ * not be @c NULL.
+ * @param[out] query A pointer to a handle that will store the created
+ * telemetry sensor query. Must not be @c NULL. The job must be freed by
```

Comment:
```suggestion
 * telemetry sensor query. Must not be @c NULL. The query must be freed by
```
That is a copy paste mistake, I guess.